encoding/encoder.go

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) bwplotka/mimic Authors
+// Licensed under the Apache License 2.0.
+
+package encoding
+
+import "io"
+
+// Encoder implements the needed functions to encode a Go struct to a particular config language.
+type Encoder interface {
+	io.Reader
+	Commenter(b []byte, comments []string) []byte
```

Comment:
I also think it's a bit confusing signature... Ideally it's consistent with Reader and it either specify in the name it's the top level, or we make it "current level".

What about simpler:

`EncodeComment(lines string) []byte` that would allow encoding one line or mutliple lines (split with \n) as comments - then those bytes can be added anywhere.

WDYT?

Reply:
Yup I agree! Will add method comment + simplify signature.

Reply:
Added method comment and simplfied it a bit. Also added testcases for comment encoding 🙂